src/components/language/ethnologue-language/ethnologue-language.service.ts

```
@@ -0,0 +1,367 @@
+    }
+
+    const result = await this.readOne(id, session);
+    return result;
+  }
+
+  async update(id: string, input: UpdateEthnologueLanguage, session: ISession) {
+    if (!input) return;
+
+    this.logger.info(`Update ethnologue language`, {
+      id,
+      input,
+      userId: session.userId,
+    });
+
+    try {
+      const query = this.db
+        .query()
+        .match(matchSession(session, { withAclRead: 'canReadLanguages' }))
+        .match([
+          node('ethnologueLanguage', 'EthnologueLanguage', {
+            active: true,
+            id: id,
+          }),
+        ])
+        .match([
+          [
+            node('ethnologueLanguage'),
+            relation('out', '', 'id', { active: true }),
+            node('id', 'Property', { active: true }),
+          ],
+          [
+            node('ethnologueLanguage'),
+            relation('out', '', 'code', { active: true }),
+            node('code', 'Property', { active: true }),
+          ],
+          [
+            node('ethnologueLanguage'),
+            relation('out', '', 'provisionalCode', { active: true }),
+            node('provisionalCode', 'Property', { active: true }),
+          ],
+          [
+            node('ethnologueLanguage'),
+            relation('out', '', 'name', { active: true }),
+            node('name', 'Property', { active: true }),
+          ],
+          [
+            node('ethnologueLanguage'),
+            relation('out', '', 'population', { active: true }),
+            node('population', 'Property', { active: true }),
+          ],
+        ])
+        .setValues({
+          'id.value': input.id,
+          'code.value': input.code,
+          'provisionalCode.value': input.provisionalCode,
+          'name.value': input.name,
+          'population.value': input.population,
+        });
+      await query.run();
+    } catch (e) {
+      this.logger.error('e:>>', e);
+    }
```

Please don't swallow errors like this.
```suggestion
    } catch (e) {
      this.logger.error('update failed', { exception: e });
      throw new ServerException('Failed to update ethnologue language');
    }
```